### PR TITLE
Document image drawing in Kiva

### LINKED ISSUE #616: Kiva: Rendering images in a kiva GC
Basic user documentation on rendering images into a GC:

- basic image drawing methods and what data they accept as input
- pixel interpolation methods (and blending? what is supported?)
- how to get image data out of a GC (eg. `save` or direct access to memory as numpy arrays/buffers, where appropriate)

### PR DESCRIPTION
**Pull request by jwiggins**
Closes #616

### CODE REVIEW

File: docs/source/kiva/images.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+Kiva Image Rendering
+====================
+
+Drawing images in kiva is accomplished via
+:py:meth:`GraphicsContext.draw_image`. A unique feature of drawing images
+(relative to path drawing) is that you can apply an arbitrary translation and
+scaling to the image without involving the current transformation matrix.
```
Comment (aaronayres35) [inline docs/source/kiva/images.rst:7]: maybe add "via the `rect` parameter"? Just to be explicit

Also very well could be unnecessary so feel free to ignore.

**Review by aaronayres35 — APPROVED** [review]
LGTM

Unrelated to this PR, but while reviewing I went through the kiva examples looking for an image one.  running the `image_explorer.py` example I saw the following error:
```
$ python image_explorer.py 
2021-03-04 12:20:39.724 Python[18846:396139] ApplePersistenceIgnoreState: Existing state will not be touched. New state will be written to (null)
Traceback (most recent call last):
  File "/Users/aayres/Desktop/enable/enable/qt4/base_window.py", line 243, in paintEvent
    self.handler.paintEvent(event)
  File "/Users/aayres/Desktop/enable/enable/qt4/base_window.py", line 78, in paintEvent
    self._enable_window._paint(event)
  File "/Users/aayres/Desktop/enable/enable/abstract_window.py", line 506, in _paint
    self.component.draw(gc, view_bounds=(0, 0, size[0], size[1]))
  File "/Users/aayres/Desktop/enable/enable/component.py", line 432, in draw
    self._draw(gc, view_bounds, mode)
  File "/Users/aayres/Desktop/enable/enable/component.py", line 813, in _draw
    self._dispatch_draw(layer, gc, view_bounds, mode)
  File "/Users/aayres/Desktop/enable/enable/component.py", line 831, in _dispatch_draw
    handler(gc, view_bounds, mode)
  File "image_explorer.py", line 40, in _draw_mainlayer
    gc.draw_image(self.image, (0, 0, gc.width(), gc.height()))
  File "/Users/aayres/Desktop/enable/kiva/agg/agg.py", line 1379, in draw_image
    return _agg.GraphicsContextArray_draw_image(self,img,rect,force_copy)
TypeError: in method 'GraphicsContextArray_draw_image', argument 2 of type 'kiva::graphics_context_base *'
Abort trap: 6
```

This probably warrants an issue?

File: docs/source/kiva/images.rst (hunk 1 above)
Comment (jwiggins) [reply docs/source/kiva/images.rst:7]: Done. I added a bit more detail about the `rect` parameter in a paragraph below.

**Comment by jwiggins** [comment]
Thanks for the feedback.

**Comment by jwiggins** [comment]
#681 added to track the image_explorer.py issue.

**Pull request merged**